zkevm-circuits/src/mpt_circuit/helpers.rs

```diff
@@ -30,8 +32,17 @@ use super::{
         get_ext_odd_nibble_value, RLPListGadget, RLPListWitness, RLPValueGadget, RLPValueWitness,
     },
     FixedTableTag,
+
+    table::Lookup,
 };
 
+// Max degree allowed in all expressions passing through the ConstraintBuilder.
+// It aims to cap `extended_k` to 2, which allows constraint degree to 2^2+1,
+// but each ExecutionGadget has implicit selector degree 3, so here it only
+// allows 2^2+1-3 = 2.
+const MAX_DEGREE: usize = 5;
+const IMPLICIT_DEGREE: usize = 3;
```

Comment:
This is EVM circuit specific so to generalize things don't use these anymore. Instead of `MAX_DEGREE` use `max_degree` as stored in the ConstraintBuilder. `IMPLICIT_DEGREE` you can drop completely, we shouldn't need it anymore.